### PR TITLE
AssessmentsTab: Multiple choice detail view v 1 / 3

### PR DESCRIPTION
**Pull request by caleybrock**
The multiple choice detail view will display the question, the answers, and what each student answered. This PR only creates the dialog and shows the question.

[Spec](https://docs.google.com/document/d/1-2-TAFVeYqIIXs1WKIgD3pNMqgUUhPZYwevFSWCHpoY/edit#heading=h.q201bj2wzlp2) for more context

I've also refactored how I showed the free response detail view to hold the state in redux instead of the component because it made it easier to use selector functions to add more to the detail view.

Storybooks added and updated, redux tests added.

screen shot 2018-07-18 at 4 41 26 pm
screen shot 2018-07-18 at 4 41 36 pm

### CODE REVIEW

File: apps/src/templates/sectionAssessments/MultipleChoiceDetailsDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    paddingLeft: 20,
+    paddingRight: 20,
+    paddingBottom: 20
+  },
+  instructions: {
+    marginTop: 20
+  }
+};
+
+class MultipleChoiceDetailsDialog extends Component {
```
Comment (caleybrock) [inline apps/src/templates/sectionAssessments/MultipleChoiceDetailsDialog.jsx:23]: This is currently very similar to the one for free response questions, but they are about to diverge significantly so I went ahead and made them separate.

File: apps/src/templates/sectionAssessments/MultipleChoiceDetailsDialog.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  static propTypes = {
+    isDialogOpen: PropTypes.bool.isRequired,
+    closeDialog: PropTypes.func.isRequired,
+    questionText: PropTypes.string,
+  };
+
+  render() {
+    // Questions are in markdown format and should not display as plain text in the dialog.
+    const renderedMarkdown = processMarkdown(this.props.questionText, { renderer });
```
Comment (caleybrock) [inline apps/src/templates/sectionAssessments/MultipleChoiceDetailsDialog.jsx:32]: This is our current standard for displaying markdown in react. Elijah's working on something safer next sprint.

File: apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js [hunk 3]
```diff
@@ -209,6 +221,30 @@ export const getCurrentAssessmentQuestions = (state) => {
...
+    const currentSurvey = surveys[state.sectionAssessments.assessmentId];
+    const questionData = currentSurvey.levelgroup_results;
+    if (questionData[state.sectionAssessments.questionIndex]) {
+      return questionData[state.sectionAssessments.questionIndex].question;
+    } else {
+      return '';
+    }
+  } else {
+    // Get question text for assessment.
+    const assessment = getCurrentAssessmentQuestions(state) || {};
```
Comment (epeach) [inline apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js:238]: I'm not familiar with the syntax `|| {}`. How does it affect the assignment to assessment?

File: apps/src/templates/sectionAssessments/SectionAssessments.jsx [hunk 4]
```diff
@@ -106,12 +107,28 @@ class SectionAssessments extends Component {
...
+    });
+  };
+
+  showMulitpleChoiceDetailDialog = () => {
+    this.setState({
+      multipleChoiceDetailDialogOpen: true
+    });
+  };
+
+  hideMultipleDetailDialog = () => {
```
Comment (epeach) [inline apps/src/templates/sectionAssessments/SectionAssessments.jsx]: Nit: Make this naming consistent with the `showMultipleChoiceDetailDialog`?

File: apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js (hunk 3 above)
Comment (caleybrock) [reply apps/src/templates/sectionAssessments/sectionAssessmentsRedux.js:238]: It just means if the first expression is falsey (undefined in this case), default `assessment` to the second expression. That way I can guarantee myself to execute the second line without error, even if there are no current assessment questions.

File: apps/src/templates/sectionAssessments/SectionAssessments.jsx (hunk 4 above)
Comment (caleybrock) [reply apps/src/templates/sectionAssessments/SectionAssessments.jsx]: Oops, definitely meant to do that.

File: apps/src/templates/sectionAssessments/MultipleChoiceDetailsDialog.jsx (hunk 1 above)
Comment (davidsbailey) [reply apps/src/templates/sectionAssessments/MultipleChoiceDetailsDialog.jsx:23]: thanks for the heads up.

File: apps/src/templates/sectionAssessments/MultipleChoiceDetailsDialog.jsx (hunk 2 above)
Comment (davidsbailey) [reply apps/src/templates/sectionAssessments/MultipleChoiceDetailsDialog.jsx:32]: erm... I meant thanks for the heads up on the i18n issue.

**Pull request merged**